pkg/handlers/default.go

```diff
@@ -49,3 +56,68 @@ func KillContainer(pid int) registry.HandlerFunc {
 		return registry.HandlerResultErrno(unix.EPERM)
 	}
 }
+
+func FreezeContainer(pid int) registry.HandlerFunc {
+	re := regexp.MustCompile(`^(\d*):(freezer)?:(/..*)$`)
+	return func(fd libseccomp.ScmpFd, req *libseccomp.ScmpNotifReq) (result registry.HandlerResult) {
+		cgroupPath := ""
+		if cgroupFile, err := os.Open(filepath.Join("/proc", fmt.Sprintf("%d", pid), "cgroup")); err == nil {
+			defer cgroupFile.Close()
+			reader := bufio.NewReader(cgroupFile)
+			for {
+				line, err := reader.ReadString('\n')
+				if err != nil {
+					break
+				}
+				line = strings.TrimSuffix(line, "\n")
+
+				r := re.FindStringSubmatch(line)
+				if len(r) < 4 {
```

Comment:
Is it guaranteed that the len(r) will be less than 4 some time? Because, otherwise, we have an infinite loop

Reply:
`reader.ReadString()` returns `io.EOF` at the end of the file, so it will not be an infinite loop.

Reply:
Cool